pandas/tests/plotting/test_hist_method.py

```diff
@@ -565,6 +565,37 @@ def test_hist_secondary_legend(self):
         assert ax.get_yaxis().get_visible()
         tm.close()
 
+    def test_hist_weights(self):
+        # GH 48884
+        # create a df and a set of weights
+        df = DataFrame(np.random.randn(10, 4), columns=list("abcd"))
+        weights = np.random.randn(len(df))
+        # set a random element of each column to NaN
+        # store remaining data and corresponding weights
+        no_nan_df = DataFrame()
```

Comment:
Can you just initialize `no_nan_df`, `df`,  `no_nan_weights` with the `NaN`s manually? It will be easier for maintainers to follow in the future.

Reply:
As opposed to choosing the NaN locations in each column randomly?

Reply:
> As opposed to choosing the NaN locations in each column randomly?

Correct.

Reply:
Thanks, but by manually, I was referring to that these objects should be constructed without and of the looping

```
df  = pd.DataFrame(...)
no_nan_df = DataFrame(...)
no_nan_weights = np.array (...)
```

Reply:
I don't see how to produce the elements of `no_nan_df` or `no_nan_weights` without looping, unless you mean to manually specify every element?

`no_nan_df` contains the same elements as `df`, but removing the NaNs from each column. `no_nan_weights` then contains the elements of `weights` repeated 4 times, but each column omitting the weight corresponding to the NaN in `df`. This is so that the histograms produced by `df` and `weights` will be the same as `no_nan_df` and `no_nan_weights`.